packages/context/src/context.ts

```diff
@@ -340,6 +350,105 @@ export class Context extends EventEmitter {
     return this;
   }
 
+  /**
+   * Create a corresponding binding for configuration of the target bound by
+   * the given key in the context.
+   *
+   * For example, `ctx.configure('controllers.MyController').to({x: 1})` will
+   * create binding `controllers.MyController:$config` with value `{x: 1}`.
+   *
+   * @param key The key for the binding to be configured
+   */
+  configure<ConfigValueType = BoundValue>(
+    key: BindingAddress<unknown> = '',
+  ): Binding<ConfigValueType> {
+    const keyForConfig = BindingKey.buildKeyForConfig(key);
+    const bindingForConfig = this.bind<ConfigValueType>(keyForConfig).tag({
+      config: key,
+    });
+    return bindingForConfig;
+  }
+
+  /**
+   * Get the value or promise of configuration for a given binding by key
+   *
+   * @param key Binding key
+   * @param configPath Property path for the option. For example, `x.y`
+   * requests for `<config>.x.y`. If not set, the `<config>` object will be
+   * returned.
+   * @param resolutionOptions Options for the resolution.
+   * - optional: if not set or set to `true`, `undefined` will be returned if
+   * no corresponding value is found. Otherwise, an error will be thrown.
+   */
+  getConfigAsValueOrPromise<ConfigValueType>(
+    key: BindingAddress<unknown>,
+    configPath?: string,
+    resolutionOptions?: ResolutionOptions,
+  ): ValueOrPromise<ConfigValueType | undefined> {
+    return this.getConfigResolver().getConfigAsValueOrPromise(
+      key,
+      configPath,
+      resolutionOptions,
+    );
+  }
+
+  private getConfigResolver() {
```

Comment:
I find it ugly to have both `configResolver` as a property and `getConfigResolver` as a getter function. Have you considered initializing `this.configResolver` right in the constructor?